lib/sm/client.rb

```diff
@@ -118,26 +118,45 @@ def delete_folder(id)
     #
     # @return [Common::Collection]
     #
-    def get_folder_messages(user_uuid, folder_id, use_cache)
+    # rubocop:disable Metrics/MethodLength
+    def get_folder_messages(user_uuid, folder_id)
       cache_key = "#{user_uuid}-folder-messages-#{folder_id}"
+
+      # MHV API doesn't allow for a page_size > total number of messages in folder, so
+      # we get the folder's message count here
+      current_folder = get_folder(folder_id)
```

Comment:
@kreek if `get_folder()` fails for here some reason, do I need to throw an exception to prevent the code below from executing, or will rails handle that for me?

Reply:
@narin dug into the code and I think it depends on the error. If it's because the request failed I think it will be handled for you, otherwise no (it will always be handled but you'll get a 500 response). However, there's a way to find out for sure 😄 Write a spec for when `get_folder` fails. Take (or make) a VCR cassette where that call fails and check to see if it's gracefully handled. If you wan to test a non HTTP error you could do something like `allow(Folder).to receive(:new).and_raise(StandardError)` before the expectation.

Reply:
Thanks. Does what I did at `spec/request/swagger_spec.rb:1148` cover the first scenario you mentioned?